### ISSUE TITLE
Networking fault injection (for testing)

### ISSUE THREAD

**Issue by xemul** [scylladb/seastar#1141]
Not all error paths in src/rpc code are checked, some do not even exist (e.g. #1137). It might make sense to add fault-injection for network communication and equip unit tests with it

**Comment by mykaul** [comment]
@xemul - anything we need to add here, or is the above commit enough to close this issue?

**Comment by xemul** [comment]
No, there's one more to come (regarding #1150). Also there's #1152 to consider
